# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by scripts/Create-CopyPatch.
# 
# T2 SDE: package/*/netkit-rsh/fixup_wait3_api_change.patch
# Copyright (C) 2021 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

Fix build issue do to Deprecate union wait and remove support from wait functions [BZ #19613]

| rlogin.c: In function 'catch_child':
| rlogin.c:463:13: error: storage size of 'status' isn't known
|   union wait status;

https://sourceware.org/ml/libc-alpha/2016-02/msg00342.html

Upstream-Status: Inappropriate [ no upstream maintaner ]

Signed-off-by Armin Kuster <akuster@mvista.com>

Index: netkit-rsh-0.17/rlogin/rlogin.c
===================================================================
--- netkit-rsh-0.17.orig/rlogin/rlogin.c
+++ netkit-rsh-0.17/rlogin/rlogin.c
@@ -460,7 +460,7 @@ writeroob(int ignore)
 void
 catch_child(int ignore)
 {
-	union wait status;
+	int status;
 	int pid;
 
 	(void)ignore;
@@ -471,7 +471,7 @@ catch_child(int ignore)
 			return;
 		/* if the child (reader) dies, just quit */
 		if (pid < 0 || (pid == childpid && !WIFSTOPPED(status)))
-			done((int)(status.w_termsig | status.w_retcode));
+			done((int)(WTERMSIG(status) | WEXITSTATUS(status)));
 	}
 	/* NOTREACHED */
 }
